### PR TITLE
Do not delete cleanup and saved_caches folders

### LINKED ISSUE #26: clean_path fails with mounted file systems
The clean path command fails if the folder to be wiped is a docker volume mount

Changing
- subprocess.check_output(['sudo', '-u', p.owner(), 'rm', '-rf', str(p)])
to
- subprocess.check_output(['sudo', '-u', p.owner(), 'rm', '-rf', str(p) + '/*'])
should allow the folder to be cleared successfully

### PR DESCRIPTION
**Pull request by arodrime**
Fixes #26, #109 an #164.
Fixes master bug in config: https://github.com/thelastpickle/cassandra-medusa/pull/168/files#diff-3feb4fc181a6e98475e3a0899f71011f
Improve tests (so they show the error better): https://github.com/thelastpickle/cassandra-medusa/pull/168/files#diff-dab28e770b45220deae663cad43f2227

### CODE REVIEW

**Comment by arodrime** [comment]
Test pass now, smoke test looks ok. Doing some more test but this can be reviewed already I think @adejanovski.

File: medusa/restore_node.py [hunk 1]
```diff
@@ -225,10 +228,19 @@ def table_is_allowed_to_restore(keyspace, table, fqtns_to_restore):
...
+def clean_path(p, keep_folder=False):
+    path = str(p)
+    if p.exists() and os.path.isdir(path) and len(os.listdir(path)):
+        logging.debug('Cleaning ({})'.format(path))
+        if keep_folder:
+            logging.debug('Removing files - keep folder {}'.format(path))
+            for f in os.listdir(path):
+                file_path = os.path.join(path, f)
+                logging.debug('Removing file {}'.format(file_path))
+                os.remove(file_path)
```
Comment (adejanovski) [inline medusa/restore_node.py:240]: Why not use the `rm` command with subprocess here as well? Or the other way around so we're consistent in how we access the FS. Either all through Python modules (more portable code) or with shell commands (easier to deal with file ownership?).

Comment (arodrime) [reply medusa/restore_node.py:240]: Makes sense, I thought about that, and dropped the ball. 100% agree on making this consistent, and we use bash commands everywhere else 👍.

Comment (arodrime) [reply medusa/restore_node.py:240]: Here is the diff, I'll make sure this works.

```
-                os.remove(file_path)
+                subprocess.check_output(['sudo', '-u', p.owner(), 'rm', '-rf', file_path])
```

**Pull request merged**